drivers/media/i2c/s5k2xx.c

```diff
@@ -0,0 +1,1468 @@
+	if (!s5k2xx)
+		return -ENOMEM;
+
+	ret = s5k2xx_of_init(s5k2xx, &client->dev);
+	if (ret)
+		return ret;
+
+	ret = s5k2xx_check_hwcfg(s5k2xx, &client->dev);
+	if (ret) {
+		dev_err(&client->dev, "failed to check HW configuration: %d",
+			ret);
+		return ret;
+	}
+
+	s5k2xx->rmap = devm_regmap_init_i2c(client, &s5k2xx_regmap_config);
+	if (IS_ERR_OR_NULL(s5k2xx->rmap)) {
+		dev_err(&client->dev, "failed to initialize regmap: %ld\n",
+			PTR_ERR(s5k2xx->rmap));
+		return PTR_ERR(s5k2xx->rmap) ?: -ENODATA;
+	}
+
+	v4l2_i2c_subdev_init(&s5k2xx->sd, client, &s5k2xx_subdev_ops);
+
+	s5k2xx_power_on(&client->dev);
+
+	ret = s5k2xx_identify_module(s5k2xx);
+	if (ret) {
+		dev_err(&client->dev, "failed to find sensor: %d", ret);
+		goto power_off;
+	}
+
+	mutex_init(&s5k2xx->lock);
+	s5k2xx->cur_mode = &s5k2xx->data->modes[0];
+	ret = s5k2xx_init_controls(s5k2xx);
+	if (ret) {
+		dev_err(&client->dev, "failed to init controls: %d", ret);
+		goto probe_error_v4l2_ctrl_handler_free;
+	}
+
+	snprintf(s5k2xx->sd.name, sizeof(s5k2xx->sd.name),
+			"%s %s", s5k2xx->data->model, dev_name(&client->dev));
+
+	s5k2xx->sd.internal_ops = &s5k2xx_internal_ops;
+	s5k2xx->sd.flags |= V4L2_SUBDEV_FL_HAS_DEVNODE;
+	s5k2xx->sd.entity.ops = &s5k2xx_subdev_entity_ops;
+	s5k2xx->sd.entity.function = MEDIA_ENT_F_CAM_SENSOR;
+	s5k2xx->pad.flags = MEDIA_PAD_FL_SOURCE;
+	ret = media_entity_pads_init(&s5k2xx->sd.entity, 1, &s5k2xx->pad);
+	if (ret) {
+		dev_err(&client->dev, "failed to init entity pads: %d", ret);
+		goto probe_error_v4l2_ctrl_handler_free;
+	}
+
+	ret = v4l2_async_register_subdev_sensor(&s5k2xx->sd);
+	if (ret < 0) {
+		dev_err(&client->dev, "failed to register V4L2 subdev: %d",
+			ret);
+		goto probe_error_media_entity_cleanup;
+	}
+
+	pm_runtime_set_active(&client->dev);
+	pm_runtime_enable(&client->dev);
+	pm_runtime_idle(&client->dev);
+
+	return 0;
+
+probe_error_media_entity_cleanup:
+	media_entity_cleanup(&s5k2xx->sd.entity);
+
+probe_error_v4l2_ctrl_handler_free:
+	v4l2_ctrl_handler_free(s5k2xx->sd.ctrl_handler);
+	mutex_destroy(&s5k2xx->lock);
+
+power_off:
+	s5k2xx_power_off(&client->dev);
+
+	return ret;
+}
+
+static const struct dev_pm_ops s5k2xx_pm_ops = {
+	SET_SYSTEM_SLEEP_PM_OPS(s5k2xx_suspend, s5k2xx_resume)
+	SET_RUNTIME_PM_OPS(s5k2xx_power_off, s5k2xx_power_on, NULL)
+};
+
+static const struct of_device_id s5k2xx_of_match[] = {
+	{ .compatible = "samsung,s5k3l8", &s5k3l8_data },
+	{ .compatible = "samsung,s5k2p6sx", &s5k2p6sx_data },
+	{ .compatible = "samsung,s5k2x7sp", &s5k2x7sp_data },
+	{ /* sentinel */ }
+};
+MODULE_DEVICE_TABLE(of, s5k2xx_of_match);
+
+static struct i2c_driver s5k2xx_i2c_driver = {
+	.driver = {
+		.name = "s5k2xx",
+		.pm = &s5k2xx_pm_ops,
+		.of_match_table	= of_match_ptr(s5k2xx_of_match),
+	},
+	.probe = s5k2xx_probe,
+	.remove = s5k2xx_remove,
+};
+
+module_i2c_driver(s5k2xx_i2c_driver);
+
+MODULE_DESCRIPTION("Samsung S5K2xx sensor driver");
+MODULE_LICENSE("GPL v2");
```

Comment:
```suggestion
MODULE_LICENSE("GPL");
```

Reply:
Same it should be modified in the source repo.